mypy/report.py

```diff
@@ -304,7 +304,7 @@ def __init__(self, reports: Reports, output_dir: str) -> None:
         self.css_html_path = os.path.join(reports.data_dir, 'xml', 'mypy-html.css')
         xsd_path = os.path.join(reports.data_dir, 'xml', 'mypy.xsd')
         self.schema = etree.XMLSchema(etree.parse(xsd_path))
-        self.last_xml = None  # type: etree._ElementTree
+        self.last_xml = None  # type: Optional[etree._ElementTree]
```

Comment:
If I understand the logic correctly, those ``None`` assignments are needed, but the attribute should be always non-``None`` at the end, so that I replaced ``if``s with asserts, I checked locally it works correctly.